dwa_local_planner/src/dwa_planner_ros.cpp

```diff
@@ -184,6 +184,42 @@ namespace dwa_local_planner {
       return false;
     }
 
+    bool reconfigure_needed = false;
+    auto limits = planner_util_.getCurrentLimits();
+
+    if (dist_tolerance > 0) {
```

Comment:
Weak S: Maybe avoid the duplicated code with something like this? :shrug: 

```cpp
std::vector<double&> latest_limits_tolerances = {latest_limits.xy_goal_tolerance, latest_limits.angle_tolerance};
std::vector<double&> limits_tolerances = {limits.xy_goal_tolerance, limits.angle_tolerance};
std::vector<double&> tolerances = {dist_tolerance, angle_tolerance};
for (size_t i = 0; i < tolerances.size(); ++i)
{
  if (tolerances[i] != limits_tolerances[i]) {
    ROS_INFO_STREAM("updating xy_goal_tolerance to tolerance from action: " << dist_tolerance);
    limits_tolerances[i] = tolerances[i];
    reconfigure_needed = true;
  } else if (latest_limits_tolenraces[i] != limits_tolerances[i]) {
    limits_tolerances[i] = _latest_limits_tolerances[i];
    reconfigure_needed = true;
  }
}
```

Reply:
hmm we'd have to parametrize the string in `ROS_INFO_STREAM` too... I think the cleanest to avoid repetition would be sth like
```
for (auto& [updated_tol, action_tol, limit_tol, name] : std::vector<std::tuple<double&, double, double, std::string>>{
        {limits.xy_goal_tolerance, dist_tolerance, _latest_limits.xy_goal_tolerance, "xy_goal_tolerance"},
        {limits.yaw_goal_tolerance, angle_tolerance, _latest_limits.yaw_goal_tolerance, "yaw_goal_tolerance" } }) {
    ...
```
but I think it's less readable honestly. 

(If it were python, we could just call `__getattribute__("xy_goal_tolerance")` on all the different objects)